compiler/semantic/proc.go

```diff
@@ -249,21 +249,40 @@ func semProc(ctx context.Context, scope *Scope, p ast.Proc, adaptor proc.DataAda
 	case *ast.Sequential:
 		return semSequential(ctx, scope, p, adaptor)
 	case *ast.Switch:
-		var cases []dag.Case
-		for k := range p.Cases {
+		var expr dag.Expr
+		if p.Expr != nil {
 			var err error
-			tr, err := semProc(ctx, scope, p.Cases[k].Proc, adaptor)
+			expr, err = semExpr(scope, p.Expr)
 			if err != nil {
 				return nil, err
 			}
-			e, err := semExpr(scope, p.Cases[k].Expr)
+		}
+		var cases []dag.Case
+		for _, c := range p.Cases {
+			var e dag.Expr
+			if c.Expr != nil {
+				var err error
+				e, err = semExpr(scope, c.Expr)
+				if err != nil {
+					return nil, err
+				}
+			} else if p.Expr == nil {
```

Comment:
Should this be just `} else {` or did you mean to check if the switch expression is nil?

Reply:
I do mean to check if the switch expression, `p.Expr`, is nil here. A nil case expression, `c.Expr`, indicates the default case, and its handling depends on whether or not there's a switch expression. I'll update the comment on the following line to make that explicit.

Reply:
Done in cdc79fd771a5e497516247cc5fa3fc133cb97293.